rs/ledger_suite/icrc1/ledger/src/lib.rs

```
@@ -619,12 +627,14 @@ fn default_decimals() -> u8 {
 }
 
 fn map_metadata_or_trap(arg_metadata: Vec<(String, Value)>) -> Vec<(String, StoredValue)> {
-    const DISALLOWED_METADATA_FIELDS: [&str; 5] = [
+    const DISALLOWED_METADATA_FIELDS: [&str; 7] = [
         METADATA_DECIMALS,
         METADATA_NAME,
         METADATA_SYMBOL,
         METADATA_FEE,
         METADATA_MAX_MEMO_LENGTH,
+        METADATA_PUBLIC_ALLOWANCES,
+        METADATA_MAX_TAKE_ALLOWANCES,
```

Thanks for adding this, I'm not sure I would have remembered! Would it perhaps make sense to at least add a comment (in the `pub fn metadata()`?) as a reminder, that if adding metadata that is computed dynamically, then users should not be allowed to specify that metadata explicitly in the init or upgrade arguments? Also, what should we do in cases where the user has previously specified some metadata that we later want to compute dynamically? Should we silently drop the user provided metadata (overwrite it with our dynamically generated one), include both in `icrc1_metadata` responses, fail the canister upgrade (forcing the user to re-set all the other metadata, including e.g., the logo), or something else? Currently, I think the implementation would include both, which can be confusing.

---

The metadata set by the user is not interpreted in any way, so for e.g., `transfer_fee`, the value set in the explicitly provided fields in the init/upgrade args are used. Whatever the user specifies in `metadata` in the init/upgrade args is purely informational.

I would suggest going with what you have here now in this PR, and follow up separately if further measures are needed?